src/libsystemd-network/dhcp6-option.c

```diff
@@ -495,6 +495,41 @@ int dhcp6_option_parse(
         return 0;
 }
 
+int dhcp6_parse_vendor_option(
```

Comment:
This is completely equivalent dhcp6_parse_option(), isn't it? Please drop this.

Reply:
It is not exactly same function as dhcp6_parse_option, it adds +4 while doing unaligned_read to get the suboption, length and data etc.

Reply:
Ah, indeed. Then, this fails when multiple suboptions in the vendor option.

Please reuse `dhcp6_parse_option()` with proper arguments.
See e.g. `dhcp6_lease_add_ntp()` or `dhcp6_option_parse_ia()` to get how to use it to parse suboptions.

Reply:
It does not fail with multiple subooptions under vendor options. I have tested this by having atleast two suboptions under vendor options passed from the server. We are shifting the offset which we have calculated in dhcp6_parse_vendor_options() function. This will just calculate the length (4 + length of vendor-option-data field as per RFC)  and then shift the offset accordingly which will point to the next suboption.
yes I used dhcp6_lease_add_ntp() as example.

Reply:
I will re-read RFC later again.

Reply:
I have read https://datatracker.ietf.org/doc/html/rfc8415#section-21.17 again. But I could not follow your point.
Why not to start loop from the beginning of the "vendor-option-data" with `dhcp6_parse_option()` ??
The "vendor-option-data" field is a normal tlv data. Please anyway use `dhcp6_parse_option()`.

Also, please introduce a simple testcase in test-dhcp6-client.c or so.

Reply:
Changed it to use dhcp6_parse_option()